src/components/hooks/useSelect.tsx

```diff
@@ -19,7 +19,9 @@ interface RenderOptions<T> {
   defaultSelectedItem?: T;
 }
 
-type SelectOptions<T> = BaseOptions<T> | RenderOptions<T>;
+type SelectOptions<T> = {
+  renderItem?: (item: T) => ReactNode;
+} & (BaseOptions<T> | RenderOptions<T>);
```

Comment:
Can you revert this change and instead make `BaseOptions` and `RenderOptions` extend a type that has `renderItem` and `defaultSelectedItem` since those 2 properties are in common?